Sources/Commands/Workspace.swift

```diff
@@ -150,6 +150,39 @@ public class Workspace {
         }
     }
 
+    /// A struct representing all the current manifests (root + external) in a package graph.
+    struct DependencyManifests {
+        /// The root manifest.
+        let root: Manifest
+
+        /// The dependency manifests in the transitive closure of root manifest.
+        let dependencies: [Manifest]
+
+        /// Computes the URLs which are declared in the manifests but aren't present in dependencies.
+        var missingURLs: Set<String> {
+            let manifestsMap = Dictionary<String, Manifest>(
+                items: [(root.url, root)] + dependencies.map{ ($0.url, $0) })
+
+            let availableURLs = Set<String>(manifestsMap.keys)
```

Comment:
Nit: I would move this below to its site of first use (this placement makes it look like it is used in the closure)